AzureFirewall/AzureFirewallApplicationRule.kql

```diff
@@ -0,0 +1,38 @@
+let applicationRuleAllowsHttp = AzureDiagnostics
+| where ResourceType == "AZUREFIREWALLS"
+| where Category == "AzureFirewallApplicationRule"
+| parse msg_s with * " request from " ip_source ":" port_source " to " ip_destination ":" port_destination ". Url: " url ". Action: " action ". Policy: " policy ". Rule Collection Group: " rulecollectiongroup "Rule Collection: " rulecollection ". Rule: " rule
+| where action == "Allow"

```

Comment:
A union parser with each union branch filtering after parsing is prohibitively slow. You need to filter on the original field (as in | where **msg_s has "Action: Allow"** two lines above). This comment is relevant for each branch.

Reply:
I think I've managed to optimize everything by combining both `where` as `parse-where` statements before perming the `union`.